cirq/circuits/circuit_test.py

```diff
@@ -109,6 +110,49 @@ def test_equality():
         ]))
 
 
+def test_approx_eq():
+
+    class TestDevice(cirq.Device):
+
+        def duration_of(self, operation: cirq.Operation) -> Duration:
+            pass
+
+        def validate_operation(self, operation: cirq.Operation) -> None:
+            pass
+
+        def validate_scheduled_operation(
+                self,
+                schedule: cirq.Schedule,
+                scheduled_operation: cirq.ScheduledOperation
+        ) -> None:
+            pass
+
+        def validate_schedule(self, schedule: 'cirq.Schedule') -> None:
+            pass
+
+    a = cirq.NamedQubit('a')
+    b = cirq.NamedQubit('b')
+
+    assert not cirq.approx_eq(Circuit([Moment([cirq.X(a)])]),
+                              Moment([cirq.X(a)]))
+
+    assert cirq.approx_eq(Circuit([Moment([cirq.X(a)])]),
+                          Circuit([Moment([cirq.X(a)])]))
+    assert not cirq.approx_eq(Circuit([Moment([cirq.X(a)])]),
+                              Circuit([Moment([cirq.X(b)])]))
+
+    assert cirq.approx_eq(Circuit([Moment([cirq.XPowGate(exponent=0)(a)])]),
+                          Circuit([Moment([cirq.XPowGate(exponent=1e-9)(a)])]))
+
+    assert not cirq.approx_eq(Circuit([Moment([cirq.XPowGate(exponent=0)(a)])]),
```

Comment:
None of these tests specify the atol property. Check that increasing the tolerance makes the equality more approximate.

Reply:
Added tests for here and in the remaining two tested files as well.